devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/ImageCheckRequirementsTask.java

```diff
@@ -0,0 +1,121 @@
+package io.quarkus.gradle.tasks;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.api.tasks.TaskAction;
+
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.gradle.tooling.ToolingUtils;
+import io.quarkus.maven.dependency.ArtifactCoords;
+
+public abstract class ImageCheckRequirementsTask extends DefaultTask {
+
+    @OutputFile
+    public abstract RegularFileProperty getOutputFile();
+
+    private static final String DEPLOYMENT_SUFFIX = "-deployment";
+    static final String QUARKUS_PREFIX = "quarkus-";
+    static final String QUARKUS_CONTAINER_IMAGE_PREFIX = "quarkus-container-image-";
+    static final String QUARKUS_CONTAINER_IMAGE_BUILD = "quarkus.container-image.build";
+    static final String QUARKUS_CONTAINER_IMAGE_PUSH = "quarkus.container-image.push";
+    static final String QUARKUS_CONTAINER_IMAGE_BUILDER = "quarkus.container-image.builder";
+
+    @InputFile
+    @PathSensitive(PathSensitivity.RELATIVE)
+    public abstract RegularFileProperty getApplicationModel();
+
+    static final Map<String, ImageCheckRequirementsTask.Builder> BUILDERS = new HashMap<>();
+    static {
+        for (ImageCheckRequirementsTask.Builder builder : ImageCheckRequirementsTask.Builder.values()) {
+            BUILDERS.put(builder.name(), builder);
+        }
+    }
+
+    enum Builder {
+        docker,
+        jib,
+        buildpack,
+        openshift
+    }
+
+    Optional<ImageCheckRequirementsTask.Builder> builderFromSystemProperties() {
+        return Optional.ofNullable(System.getProperty(QUARKUS_CONTAINER_IMAGE_BUILDER))
+                .filter(BUILDERS::containsKey)
+                .map(BUILDERS::get);
+    }
+
+    List<ImageCheckRequirementsTask.Builder> availableBuilders() throws IOException {
+        // This will only pickup direct dependencies and not transitives
```

Comment:
The way it looks it will iterate over all the dependencies, not just the direct ones.

Reply:
I think you're right, the dependencies used here are the ones set in the ApplicationModel(through the `QuarkusApplicationModelTask`).
The original code(that was implemented in the `ImageTask`) was:
```
        // This will only pickup direct dependencies and not transitives
        // This means that extensions like quarkus-container-image-openshift via quarkus-openshift are not picked up
        // So, let's relax our filters a bit so that we can pickup quarkus-openshift directly (relax the prefix requirement).
        return getProject().getConfigurations()
                .getByName(ToolingUtils.toDeploymentConfigurationName(
                        ApplicationDeploymentClasspathBuilder.getFinalRuntimeConfigName(LaunchMode.NORMAL)))
                .getDependencies().stream()
```
So I understand `appModel.getDependencies()` include dependencies and extension dependencies. I could filter by deployment flags, what do you think?

Reply:
Those already are all the deployment dependencies. So perhaps simply the comment needs to be adjusted?

Reply:
881b93757db4b7b1d95f0b5494ff7548fc774da1